mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp

```diff
@@ -43,9 +43,13 @@ using namespace mlir;
 template <typename LabelT>
 static LogicalResult checkExtensionRequirements(
     LabelT label, const spirv::TargetEnv &targetEnv,
-    const spirv::SPIRVType::ExtensionArrayRefVector &candidates) {
+    const spirv::SPIRVType::ExtensionArrayRefVector &candidates,
```

Comment:
Instead of adding this new parameter, could we pass in only a subset of candidates to this function? IE prune it on the caller side.

Reply:
Yes, that is also possible. Wanted to keep the signature of all 3 of the requirement checking function similar.
Also, wanted to have the prunning logic in one place.

Should I change the signature of this function?